### PR TITLE
Fix clientset generation for resource names matching Go keyword

### PR DESCRIPTION
**Pull request by tamalsaha**
Signed-off-by: Tamal Saha <tamal@appscode.com>

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

clientset generator will fail is resource type name matches Go keyword. This pr fixes the generated client to use `keyword_` style name in those cases.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/code-generator/hack/verify-codegen.sh [hunk 1]
```diff
@@ -49,7 +49,8 @@ fi
 
 # smoke test
 echo "Smoke testing examples by compiling..."
-go build "./${SCRIPT_ROOT}/examples/crd/..."
-go build "./${SCRIPT_ROOT}/examples/apiserver/..."
-go build "./${SCRIPT_ROOT}/examples/MixedCase/..."
-go build "./${SCRIPT_ROOT}/examples/HyphenGroup/..."
+cd ${SCRIPT_ROOT}/examples
```
Comment (tamalsaha) [inline staging/src/k8s.io/code-generator/hack/verify-codegen.sh:52]: This is needed because there is an internal go.mod file now.

File: staging/src/k8s.io/code-generator/hack/verify-codegen.sh [hunk 2]
```diff
@@ -49,7 +49,9 @@ fi
...
-go build "./${SCRIPT_ROOT}/examples/crd/..."
-go build "./${SCRIPT_ROOT}/examples/apiserver/..."
-go build "./${SCRIPT_ROOT}/examples/MixedCase/..."
-go build "./${SCRIPT_ROOT}/examples/HyphenGroup/..."
+cd "${SCRIPT_ROOT}"/examples
+go build "./crd/..."
+go build "./apiserver/..."
+go build "./MixedCase/..."
+go build "./HyphenGroup/..."
+go build "./keyword/..."
```
Comment (sttts) [inline staging/src/k8s.io/code-generator/hack/verify-codegen.sh:57]: why this change?

**Comment by sttts** [comment]
Lgtm. Only question about script change.

**Comment by sttts** [comment]
@tamalsaha ping

**Comment by dims** [comment]
Is this PR still needed, please rebase if so (or we can close it?)

**Comment by tamalsaha** [comment]
@sttts , I have rebased the pr and removed that script change. PTAL.

**Comment by tamalsaha** [comment]
@sttts , all tests are passing now. PTAL.

**Comment by sttts** [comment]
`examples/keyword/clientset/versioned/typed/default/v1/fake/fake_interface_.go` filename looks wrong

**Comment by sttts** [comment]
Can you split the code changes from the generated code? Make it hard to review in Github.
